client-misk-jooq/src/test/kotlin/app/cash/backfila/client/misk/MiskJooqBackfillTests.kt

```diff
@@ -0,0 +1,497 @@
+      val run = backfillOption.createDryRun(
+        rangeEnd = backfillOption.backfillRowKeys[0].toString()
+      )
+        .apply { configureForTest() }
+      assertThat(run.rangeEnd).isEqualTo(backfillOption.backfillRowKeys[0].toString())
+      app.cash.backfila.client.misk.testing.assertThat(run.partitionProgressSnapshot.values.single())
+        .isNotDone()
+
+      run.singleScan()
+      assertThat(run.batchesToRunSnapshot.single().utf8RangeStart()).isEqualTo(
+        backfillOption.backfillRowKeys[0].toString()
+      )
+      assertThat(run.batchesToRunSnapshot.single().matchingRecordCount).isEqualTo(1)
+      assertThat(run.batchesToRunSnapshot.single().scannedRecordCount).isEqualTo(1)
+      run.runBatch()
+
+      run.singleScan()
+      app.cash.backfila.client.misk.testing.assertThat(run).hasNoBatchesToRun()
+      app.cash.backfila.client.misk.testing.assertThat(run).isFinishedScanning()
+
+      app.cash.backfila.client.misk.testing.assertThat(run).isComplete()
+      assertThat(run.backfill.idsRanDry).containsExactly(backfillOption.backfillRowKeys[0])
+      assertThat(run.backfill.idsRanWet).isEmpty()
+    }
+  }
+
+  @TestFactory
+  fun twoBatchesOf10ToGet20Records() = listOf(
+    JooqBackfillTestOptions(
+      backfillClass = JooqMenuTestBackfill::class,
+      backfillRowKeys = JooqMenuBackfillDbSetup.createSome(transacter),
+      backfila
+    ),
+    JooqBackfillTestOptions(
+      backfillClass = JooqWidgetCompoundKeyBackfill::class,
+      backfillRowKeys = WidgetCompoundKeyBackfillDbSetup.createSome(transacter, clock, tokenGenerator),
+      backfila
+    )
+  ).map { backfillOption ->
+    DynamicTest.dynamicTest("twoBatchesOf10ToGet20Records test ${backfillOption.backfillClass.simpleName}") {
+      val run = backfillOption.createDryRun()
+        .apply { configureForTest() }
+      run.precomputeRemaining()
+
+      run.singleScan()
+      assertThat(run.batchesToRunSnapshot.single().utf8RangeStart()).isEqualTo(
+        backfillOption.backfillRowKeys[0].toString()
+      )
+      assertThat(run.batchesToRunSnapshot.single().utf8RangeEnd()).isEqualTo(
+        backfillOption.backfillRowKeys[9].toString()
+      )
+      assertThat(run.batchesToRunSnapshot.single().matchingRecordCount).isEqualTo(10)
+      assertThat(run.batchesToRunSnapshot.single().scannedRecordCount).isEqualTo(10)
+      run.runBatch()
+      app.cash.backfila.client.misk.testing.assertThat(run).hasNoBatchesToRun()
+
+      run.singleScan()
+      assertThat(run.batchesToRunSnapshot.single().utf8RangeEnd()).isEqualTo(
+        backfillOption.backfillRowKeys[19].toString()
+      )
+      assertThat(run.batchesToRunSnapshot.single().matchingRecordCount).isEqualTo(10)
+      assertThat(run.batchesToRunSnapshot.single().scannedRecordCount).isEqualTo(
+        15
+      ) // Skipped 5 records in between
+      run.runBatch()
+      app.cash.backfila.client.misk.testing.assertThat(run).hasNoBatchesToRun()
```

Comment:
static import?

Reply:
had to introduce an import alias.. miskTestingAssertThat